kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/TaxReporting1099ServiceImpl.java

```diff
@@ -0,0 +1,1806 @@
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("attempting to load 1099 form for payee= " + id + ", year=" + year);
+		}
+
+		// make sure we have a default payer
+		Payer payer = getDefaultPayer();
+		if (payer == null) {
+			throw new RuntimeException("Default payer for 1099 process not found. Check 1099 payer system property configuration.");
+		}
+
+		Integer taxYear = Integer.valueOf(TaxHelper.getTaxYear(parameterService));
+
+		if ((year != null) && StringUtils.isNumeric(year)) {
+			taxYear = Integer.valueOf(year);
+		}
+
+		PdfReader reader = getPdfReader(taxYear.intValue());
+
+		getTaxThreshold(true);
+		getTaxAmountByPaymentType(true);
+
+		// write out populated pdf 1099 forms
+		try {
+			List<Payee> payees = new ArrayList<Payee>();
+			Payee p = loadPayee(taxYear, getPayee(id));
+
+			if (p != null) {
+				payees.add(p);
+				createPdfFile(reader, TaxPropertyConstants.PDF_1099_PAGE_COPY_A, bos, payer, payees);
+			} 
+			else {
+				throw new RuntimeException("cannot find Payee: " + id);
+			}
+		} 
+		catch (Exception ex) {
+			throw new RuntimeException("error creating 1099 form for payee: " + id, ex);
+		} 
+		finally {
+			try {
+				reader.close();
+			} catch (Exception ex) {
+			}
+		}
+
+		return bos.toByteArray();
+	}
+
+	@Override
+	public void updatePayeeInformation(VendorDetail vendorDetail) {
+		if (vendorDetail.isVendorParentIndicator()) {
+			// lets get the parent payee for this vendor if it exists
+			Integer taxYear = TaxHelper.getTaxYear(parameterService);
+
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("vendorNumber: " + vendorDetail.getVendorNumber() + ", taxYear: " + taxYear);
+			}
+
+			Payee payee = getPayeeByVendorNum(vendorDetail.getVendorNumber(), taxYear);
+
+			// if we found the parent lets get the vendor detail and update the payee
+			if (payee != null) {
+				if (LOG.isDebugEnabled()) {
+					LOG.debug("payee: " + payee.getId());
+				}
+
+				String vendorName = null;
+
+				if (vendorDetail.isVendorFirstLastNameIndicator() && StringUtils.isNotBlank(vendorDetail.getVendorLastName()) && StringUtils.isNotBlank(vendorDetail.getVendorFirstName())) {
+					vendorName = (vendorDetail.getVendorLastName() + VendorConstants.NAME_DELIM + vendorDetail.getVendorFirstName());
+				} 
+				else {
+					vendorName = vendorDetail.getVendorName();
+				}
+
+				if (StringUtils.isNotBlank(vendorName)) {
+					if (vendorName.length() > VendorConstants.VENDOR_FIRST_PLUS_LAST_NAME_MAXLENGTH) {
+						payee.setVendorName(vendorName.substring(0, VendorConstants.VENDOR_FIRST_PLUS_LAST_NAME_MAXLENGTH));
+					} 
+					else {
+						payee.setVendorName(vendorName);
+					}
+				}
+
+				payee.setHeaderTaxNumber(vendorDetail.getVendorHeader().getVendorTaxNumber());
+
+				if (LOG.isDebugEnabled()) {
+					LOG.debug("vendor name: " + vendorDetail.getVendorName());
+				}
+
+				VendorAddress address = getVendorDefaultAddress(vendorDetail);
+
+				if (address != null) {
+					setPayeeAddress(address, payee);
+
```

Comment:
Please keep as is. This is an artifact of utilizing a repository design pattern that is non-generic. To genericize would be too big of a refactor.